docs: Improve selection of builder (#2512)
slsa-framework/slsa-github-generator: README.md

  - [Hall of Fame](#hall-of-fame)
    - [Generation of Provenance](#generation-of-provenance)
    - [Builder Creation](#builder-creation)
- [Generate provenance](#generate-provenance)
  - [Referencing SLSA builders and generators](#referencing-slsa-builders-and-generators)
  - [Builders](#builders)
  - [Generators](#generators)
- [Verify provenance](#verify-provenance)
  - [Installation](#installation)
  - [Inputs](#inputs)
  - [Command line examples](#command-line-examples)
- [Build Your Own Builder](#build-your-own-builder)

> - [Roadmap](#roadmap)

Reviewer: ```suggestion
- [Project Roadmap](#project-roadmap)
```

## Overview

> This repository contains free tools to generate and verify SLSA3+ provenance for native GitHub projects using GitHub Actions.

Reviewer: nit: should we use the SLSA v1 terminology?
```suggestion
This repository contains free tools to generate and verify SLSA Build Level 3 provenance for native GitHub projects using GitHub Actions.
```

This repository hosts the following builders:

| Ecosystem |      Builder      |  Description | Status |

> |:-----------|:-----------------|:------------|:--------|

Reviewer 2: Beautiful!

| [Go](https://go.dev/) projects | [Go Builder](internal/builders/go/README.md) | Builds and generates provenance for Go projects | [available since v1.0.0](https://github.com/slsa-framework/slsa-github-generator/milestone/1) |

> | [nodejs](https://nodejs.org) projects | [Node.js Builder](internal/builders/nodejs/README.md)   |  Builds and generates provenance for npm packages | [Beta since v1.6.0](https://github.com/slsa-framework/slsa-github-generator/milestone/8). [Expected GA release Sept 2023](https://github.com/slsa-framework/slsa-github-generator/milestone/17)

Reviewer: ```suggestion
| [Node.js](https://nodejs.org) projects | [Node.js Builder](internal/builders/nodejs/README.md)   |  Builds and generates provenance for npm packages | [Beta since v1.6.0](https://github.com/slsa-framework/slsa-github-generator/milestone/8). [Expected GA release Sept 2023](https://github.com/slsa-framework/slsa-github-generator/milestone/17)
```

| [Maven](https://maven.apache.org/) projects | [Maven builder](internal/builders/maven/README.md) | Build Maven packages and generates provenance. Can be uploaded to [Maven central](https://search.maven.org) | [Beta since v1.8.0](https://github.com/slsa-framework/slsa-github-generator/milestone/14) |
| [Gradle](https://gradle.org/) projects | [Gradle builder](internal/builders/gradle/README.md) | Build Gradle projects and generates provenance. Can be uploaded to [Maven central](https://search.maven.org) | [Beta since v1.8.0](https://github.com/slsa-framework/slsa-github-generator/milestone/15) |

> | [JReleaser](https://jreleaser.org/) projects | [JReleaser builder](https://github.com/jreleaser/release-action/tree/java#slsa-builder) | Builds and generates provenance using [JReleaser](https://jreleaser.org/) | [Beta since v1.8.0](https://github.com/slsa-framework/slsa-github-generator/milestone/16) |

Reviewer: Should this be here? it looks like a duplicate of the JReleaser links below?
```suggestion
```

| [docker](https://www.docker.com/) images | Container Builder | Builds docker containers and generates provenance. The generated provenance is compatible with [cosign](https://github.com/sigstore/cosign)'s attestation format | [WIP](https://github.com/slsa-framework/slsa-github-generator/milestone/5)  |
| Any | [Container-based Builder](internal/builders/docker/README.md) | Builds projects whose build pipeline is defined with a Dockerfile | [Beta since v1.7.0](https://github.com/slsa-framework/slsa-github-generator/milestone/16) |

There are other available builders using this repository's [BYOB framework](#build-your-own-builder) and not hosted in this repository:

| Ecosystem |      Builder      |  Description | Status |
|:-----------|:-----------------|:------------|:--------|
| [JReleaser](https://jreleaser.org/) projects | [JReleaser builder](https://github.com/jreleaser/release-action/tree/java#slsa-builder) | Builds and generates provenance using [JReleaser](https://jreleaser.org/) | [Beta since v1.8.0](https://github.com/slsa-framework/slsa-github-generator/milestone/16) |

> 
> 

Reviewer: Fix extra blank line
```suggestion

```

If none of these options fit your needs, use a [generator](#generators) as described below:

### Generators

> Generators only generate provenance for you. They let you meet the [provenance generation](https://slsa.dev/spec/v1.0/requirements#provenance-generation) requirements

Reviewer 2: I think some clarification would help: "They let you [nit: your artifact?] meet the provenance generation requirements for SLSA Build Level 3, but do not provide the [isolated build environment? hardened build environment?] required to fulfill the other requirements for Build Level 3."

Author: removed this sentence entirely because v1.0 makes no distinction between provenance vs build requirements. the build requirements are still met by the GitHub runner. Once https://github.com/slsa-framework/slsa-github-generator/issues/1868 is resolved we will be good to go.

for [SLSA Build level 3](https://slsa.dev/spec/v1.0/levels).

Generators create an attestation to a software artifact coming from your repository.

This repository hosts the following generators:

| Artifact type |      Generator    |  Description | Status |
|:--------------|:------------------|:-------------|:-------|

> | file | [Generic generator](internal/builders/generic/README.md) | Generates provenance for arbitrary file-based artifacts, for any ecosystem and programming language | [available since v1.2.0](https://github.com/slsa-framework/slsa-github-generator/milestone/2) |

Reviewer: nit: add examples
```suggestion
| file (binary, package tarball etc.) | [Generic generator](internal/builders/generic/README.md) | Generates provenance for arbitrary file-based artifacts, for any ecosystem and programming language | [available since v1.2.0](https://github.com/slsa-framework/slsa-github-generator/milestone/2) |
```

| container | [Container generator](internal/builders/container/README.md)   |  Generate provenance for container images. The generated provenance is compatible with    [cosign](https://github.com/sigstore/cosign)'s attestation format. | [available since v1.4.0](https://github.com/slsa-framework/slsa-github-generator/milestone/3)

## Verify provenance

> To verify the provenance, use the [github.com/slsa-framework/slsa-verifier](https://github.com/slsa-framework/slsa-verifier) project.

Reviewer 2: ```suggestion
To verify provenance created by any of the builders in this repository, use the [github.com/slsa-framework/slsa-verifier](https://github.com/slsa-framework/slsa-verifier) project.
```
Added a clarification -- is is correct?

## Build Your Own Builder

> Use the [BYOB framework](BYOB.md) to create your own SLSA3 builder on GitHub. If you have an existing GitHub Action, you can use the BYOB framework to wrap it into a SLSA3 builder.

Reviewer 3: ```suggestion
Use the [BYOB framework](BYOB.md) to create your own SLSA builder on GitHub. If you have an existing GitHub Action, you can use the BYOB framework to wrap it into a SLSA builder.
```

> If you want to build your own builder, use the [BYOB framework](BYOB.md). The framework lets you create your own SLSA3 builder on GitHub. For example, you can wrap an existing GitHub Action into a SLSA3 builder. For verification, your users can use the [slsa-verifier](#verification-of-provenance).

Reviewer 2: ```suggestion
Use the [BYOB framework](BYOB.md) to create your own SLSA3 builder on GitHub. If you have an existing GitHub Action, you can use the BYOB framework to wrap it into a SLSA3 builder. This will harden the build process to meet Build Level 3 expectations and produce Build Level 3 provenance.  To verify the provenance, your users can use the [slsa-verifier](#verification-of-provenance).
```
Made some suggestions. I'm not sure "wrap it into a SLSA3 Builder is self-explanatory, but I'll let you decide on that.

> ## Roadmap

Reviewer: ```suggestion
## Project Roadmap
```
